src/main/java/com/opentable/db/postgres/embedded/EmbeddedPostgres.java

```diff
@@ -14,152 +14,100 @@
-        this.port = port;
-        this.pgDir = pgDirectoryResolver.getDirectory(overrideWorkingDirectory);
-        this.errorRedirector = errorRedirector;
-        this.outputRedirector = outputRedirector;
-        this.pgStartupWait = Objects.requireNonNull(pgStartupWait, "Wait time cannot be null");
-        if (parentDirectory != null) {
-            mkdirs(parentDirectory);
-            cleanOldDataDirectories(parentDirectory);
-            if (dataDirectory != null) {
-                this.dataDirectory = dataDirectory;
-            } else {
-                this.dataDirectory = new File(parentDirectory, instanceId.toString());
-            }
-        } else {
-            this.dataDirectory = dataDirectory;
-        }
-        if (this.dataDirectory == null) {
-            throw new IllegalArgumentException("no data directory");
-        }
-        LOG.debug("{} postgres: data directory is {}, postgres directory is {}", instanceId, this.dataDirectory, this.pgDir);
-        mkdirs(this.dataDirectory);
 
-        lockFile = new File(this.dataDirectory, LOCK_FILE_NAME);
 
-        if (cleanDataDirectory || !new File(dataDirectory, "postgresql.conf").exists()) {
-            initdb();
+    EmbeddedPostgres(Map<String, String> postgresConfig,
+                     Map<String, String> localeConfig,
+                     DockerImageName image
+    ) throws IOException {
+        this(postgresConfig, localeConfig, image,  DEFAULT_PG_STARTUP_WAIT);
+    }
+
+    EmbeddedPostgres(Map<String, String> postgresConfig,
+                     Map<String, String> localeConfig,
+                     DockerImageName image,
+                     Duration pgStartupWait
+    ) throws IOException {
+        this.postgreDBContainer = new PostgreSQLContainer<>(image)
+                .withDatabaseName(POSTGRES)
+                .withUsername(POSTGRES)
+                .withPassword(POSTGRES)
+                .withStartupTimeout(pgStartupWait)
+                .withLogConsumer(new Slf4jLogConsumer(LOG))
+                .withEnv("POSTGRES_INITDB_ARGS", String.join(" ", createInitOptions(localeConfig)))
+                .withEnv("POSTGRES_HOST_AUTH_METHOD", "trust");
+        final List<String> cmd = new ArrayList<>(Collections.singletonList(POSTGRES));
+        cmd.addAll(createConfigOptions(postgresConfig));
+        postgreDBContainer.setCommand(cmd.toArray(new String[0]));
+        postgreDBContainer.start();
+    }
+
+    private List<String> createConfigOptions(final Map<String, String> postgresConfig) {
+        final List<String> initOptions = new ArrayList<>();
+        for (final Map.Entry<String, String> config : postgresConfig.entrySet()) {
+            initOptions.add("-c");
+            initOptions.add(config.getKey() + "=" + config.getValue());
         }
+        return initOptions;
+    }
 
-        lock();
-        startPostmaster(connectConfig);
+    private List<String> createInitOptions(final Map<String, String> localeConfig) {
```

Comment:
This is confusing. Note under createConfigIOptions we have a hasmap called initOptions and under createInitOptions we have a localeOptions. 

Also what is the distinction?